### PR TITLE
Add http/reactive_routes modules

### PR DESCRIPTION
**Pull request by pjgg**
Reactive routes migration

Migrate Reactive routes from [Beefy-ts ](https://github.com/quarkus-qe/beefy-scenarios) to quarkus-test-suite

FIX GH-96

### CODE REVIEW

**Comment by Sgitario** [comment]
Pls, rebase the PR.

**Comment by Sgitario** [comment]
Pls, add the issue link to be autoclosed once merged.

**Comment by pjgg** [comment]
> Pls, add the issue link to be autoclosed once merged.

GH-96

File: BEEFY_TS_MIGRATION_STATUS.md [hunk 1]
```diff
@@ -4,4 +4,5 @@
 | 012-quarkus-kafka-streams | messaging/kafka-streams-reactive-messaging |
 | 013-quarkus-oidc-restclient | security/keycloak-oidc-client |
 | 020-quarkus-http-non-application-endpoints | http/http-advanced |
-| 021-quarkus-panache-multiple-pus | sql-db/multiple-pus |
\ No newline at end of file
+| 304-quarkus-vertx-routes | http/reactive-routes |
```
Comment (Sgitario) [inline BEEFY_TS_MIGRATION_STATUS.md:7]: Pls, put this line in order so it's more readable.

File: http/reactive-routes/pom.xml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>io.quarkus.ts.qe</groupId>
+        <artifactId>parent</artifactId>
+        <version>1.0.0-SNAPSHOT</version>
+        <relativePath>../..</relativePath>
+    </parent>
+    <artifactId>reactive-routes</artifactId>
+    <packaging>jar</packaging>
+    <name>Quarkus QE TS: reactive-routes</name>
```
Comment (Sgitario) [inline http/reactive-routes/pom.xml:12]: I think it's clearer to name it <name>Quarkus QE TS: HTTP: reactive-routes</name>.
In order to follow the same name convention with the rest of HTTP modules, what about to change the artifactId to http-reactive-routes too?

File: http/reactive-routes/src/test/java/io/quarkus/ts/BasicsRouteHandlerIT.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.test.bootstrap.RestService;
+import io.quarkus.test.scenarios.QuarkusScenario;
+import io.quarkus.test.services.QuarkusApplication;
+
+@QuarkusScenario
+public class BasicsRouteHandlerIT {
+
+    @QuarkusApplication(ssl = true)
```
Comment (Sgitario) [inline http/reactive-routes/src/test/java/io/quarkus/ts/BasicsRouteHandlerIT.java:13]: why ssl true?

File: http/reactive-routes/src/test/java/io/quarkus/ts/validation/ValidationOnRequestBodyRouteHandlerIT.java [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+import static io.restassured.RestAssured.given;
+
+import org.apache.http.HttpStatus;
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.test.junit.QuarkusTest;
+import io.quarkus.ts.validation.utils.ValidationErrorResponse;
+import io.vertx.core.json.Json;
+
+@QuarkusTest
```
Comment (Sgitario) [inline http/reactive-routes/src/test/java/io/quarkus/ts/validation/ValidationOnRequestBodyRouteHandlerIT.java:16]: Do you leave this annotation on purpose?

**Review by Sgitario — CHANGES_REQUESTED** [review]

File: http/reactive-routes/src/test/java/io/quarkus/ts/BasicsRouteHandlerIT.java (hunk 3 above)
Comment (pjgg) [reply http/reactive-routes/src/test/java/io/quarkus/ts/BasicsRouteHandlerIT.java:13]: I think that was something that was left over. In this scenario we are not testing anything related to http2 so..I think that could be removed

**Review by Sgitario — APPROVED** [review]
lgtm!
let's wait for the OpenShift run.

**Comment by Sgitario** [comment]
run tests

**Comment by Sgitario** [comment]
@pjgg can you rebase and resolve the conflicts?

**Pull request merged**